shell/src/main/java/alluxio/shell/command/FileInfoCommand.java

```diff
@@ -58,12 +57,34 @@ protected void runCommand(AlluxioURI path, CommandLine cl) throws AlluxioExcepti
   }
 
   @Override
+  protected void runCommand(AlluxioURI path, CommandLine cl) throws AlluxioException, IOException {
+    URIStatus status = mFileSystem.getStatus(path);
+    if (status.isFolder()) {
+      List<URIStatus> childrens = mFileSystem.listStatus(path);
```

Comment:
I feel like this does not have to list the children. It should only print out the details of the directory. Listing of the children can already be done by the shell.